lib/stats/stats-registry.c

```diff
@@ -173,6 +214,13 @@ stats_register_counter(gint stats_level, const StatsClusterKey *sc_key, gint typ
   return _register_counter(stats_level, sc_key, type, FALSE, counter);
 }
 
+StatsCluster *
+stats_register_external_counter(gint stats_level, const StatsClusterKey *sc_key, gint type,
```

Comment:
I don't want to hide this... and I can imagine that someone wants to store the `StatsCounterItem` itself (for passing it to other layers as a read-only 'view').

On the other hand I can imagine that someone would require a more sophisticated API - I'll address it in another PR.

Reply:
Ok, fine for me. A small request: if a caller is not interested in the stats `StatsCounterItem`, it would be nice if they could pass `NULL` to the output parameter.. Thus making the output optional.

Reply:
Based on your request I _removed_ the output parameter.
Now you want me to add it again and enable users to set it to NULL? (I won't do that :) )

Reply:
Oh right, I reviewed an older version. It looks neat.